offload_data/plugin.cpp

```diff
@@ -0,0 +1,187 @@
+#include "common/plugin.hpp"
+#include "common/threadloop.hpp"
+#include "common/switchboard.hpp"
+#include "common/phonebook.hpp"
+#include "common/data_format.hpp"
+#include "common/global_module_defs.hpp"
+
+#include <iomanip>
+#include <fstream>
+#include <numeric>
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include "common/gl_util/lib/stb_image_write.h"
+
+using namespace ILLIXR;
+
+std::string
+getenv_or(std::string var, std::string default_) {
```

Comment:
This function is used multiply times by timewarp_gl, offload_data, and pose-lookup. It should be put somewhere in common.

I saw a comment in timewarp_gl that says we will use #198 instead of this, so I temporally copied this function to offload_data and pose-lookup.